src/components/CommentInputBox.jsx

```diff
@@ -0,0 +1,59 @@
+import { postComment } from '../api';
+import { useState, useContext } from 'react';
+import { UserContext } from '../contexts/LoggedInUser';
+
+export default function CommentInputBox({ refreshComments, article_id }) {
+
+    const [ commentBody, setComment ] = useState('');
+    const { submitComment, isLoading, error } = sendComment(refreshComments, article_id);
+    const { user } = useContext(UserContext);
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        submitComment(user, commentBody);
+        setComment('');
+    }
+
+    if (isLoading) return <div className='comment-input-box'>Loading...</div>;
+    if (error) return <div className='comment-input-box'>Error: {error.msg}</div>;
+
+    return (
+    <>
+        <form className='comment-input-box' onSubmit={handleSubmit}>
+            <label htmlFor='inputbox'>Comment </label>
+            <input id='inputbox' value={commentBody} onChange={(event) => setComment(event.target.value)}></input>
+            <button>Submit</button>
+        </form>
+    </>
+    )
+}
+
+function sendComment(refreshComments, article_id) {
+
+    const [isLoading, setLoading] = useState(false);
+    const [error, setError] = useState(null);
+
+    const sendComment = async (commentToPost) => {
+        try {
+            setLoading(true);
+            const comment = await postComment(article_id, commentToPost);
+            console.log(comment);
+        } catch (err) {
+            setError(err);
+        } finally {
+            setLoading(false);
+            refreshComments();
```

Comment:
Following this code I think what is happening here is you're making another request (via `refreshComments` to re-fetch the comments after posting one and then updating the state based on this re-fetched info.
Though this works, it adds an unnecessary API call into the process, and re-renders the whole page. Instead we could just update the state of the comment list to include our newly added comment, without relying on another call to the API